source/text.cpp

```diff
@@ -469,6 +469,18 @@ spv_result_t encodeInstructionStartingWithImmediate(
   return SPV_SUCCESS;
 }
 
+// If |str| denotes a number between 0 and 65535, then return that number.
+// Otherwise return 0.
+static int As16BitNumber(const char* str) {
+  int result = 0;
+  char c;
+  // Succeed only if we parse one parameter
+  if ((1 == std::sscanf(str, "%u%c", &result, &c)) && result <= 0xffff)
```

Comment:
%u requires an unsigned int pointer and would also likely succeed on a negative number.

Consider
if ((1 == std::sscanf(str, "**%d**%c", &result, &c)) && **result >= 0** && result <= 0xffff)